qiskit_experiments/framework/base_experiment.py

```diff
@@ -267,13 +267,10 @@ def run(
         run_opts = run_opts.__dict__
 
         # Generate and transpile circuits
-        transpile_opts = copy.copy(experiment.transpile_options.__dict__)
-        transpile_opts["initial_layout"] = list(experiment.physical_qubits)
-        circuits = transpile(experiment.circuits(), experiment.backend, **transpile_opts)
-        experiment._postprocess_transpiled_circuits(circuits, **run_options)
+        transpiled_circuits = self._transpile(experiment.circuits(), experiment.backend)
```

Comment:
I find it confusing that `self` and `experiment` exist and are being used side-by-side. In the current code, from the moment that we define `experiment`, we forget about `self` and stick with `experiment` until the end of the method. I'm not fond of this situation, because it still calls for bugs related to someone not noticing this fact and using `self` where it should be `experiment` or vice versa, but I can live with it. But if they're really both used, then it's even more confusing. So, if you can replace `self` here with `experiment`, then please do. Otherwise, it's important to document exactly why the mixture is necessary. And the best is, of course, if there is a way to get rid of `experiment` in the first place. But I don't see such a way. Is it that important to support setting of backend and analysis in run time?